### DISCUSSION TITLE
Calendar with timezone and time selection

### DISCUSSION THREAD

**Discussion by MaW000** [shadcn-ui/ui#836] [Q&A]
https://gist.github.com/fernandops26/da681c4b12e52191803b4fcb040cdebb

I copied this and it looks / works well. I'm wondering if anyone has tips on how I can implement a time zone dropdown to this without hardcoding all the time zones into a array and mapping it into a dropdown.

**Comment by huybuidac** [comment]
For anyone landing here from search — if you need timezone-aware datetime selection (IANA TZ, not just UTC offset), here's a working approach using `react-day-picker` v9's `TZDate`:

```tsx
import { DateTimePicker } from '@/components/datetime-picker';

const [date, setDate] = useState<Date | undefined>();

<DateTimePicker
  value={date}
  onChange={setDate}
  timezone="America/New_York"
  min={subHours(new Date(), 2)}
  max={addMonths(new Date(), 2)}
/>
```

It also handles segmented keyboard input and IME composition (Android Gboard / iOS soft-keyboard) — the bits that are tricky to get right.

Full source + shadcn CLI install: https://github.com/huybuidac/shadcn-datetime-picker

The CLI install is one command:
`npx shadcn@latest add https://shadcn-datetime-picker-pro.vercel.app/r/datetime-picker.json`

— and it drops the source into your `components/` folder so you own the code (no runtime dependency on the repo).
